packages/atlas/src/providers/overlayManager.tsx

```diff
@@ -78,7 +78,11 @@ export const useOverlayManager = () => {
 
   const decrementOverlaysOpenCount = useCallback(() => {
     setOverlaysSet((prevSet) => {
-      prevSet.delete(overlayId)
+      if (prevSet.size === 1) {
+        prevSet.clear()
+      } else {
+        prevSet.delete(overlayId)
+      }
```

Comment:
I'm afraid this may have some undesired consequences that we may not foresee right now, because it changes the mechanics of the manager. Take a look at this code I've added:
https://github.com/Joystream/atlas/blob/6355b6bb24e8b271c369a346ca8e31bc3c6bf62d/packages/atlas/src/components/_navigation/SidenavBase/SidenavBase.tsx#L51-L57

I think if we just added this to the `useEffect`, it would fix the bug?
```tsx
return decrementOverlaysOpenCount
```

Reply:
@kdembler Yes it would do the job but I think my solution is more versatile, we won't have to worry about handling this case in the future.